demosplan/DemosPlanCoreBundle/Logic/Import/Statement/StatementSpreadsheetImporterWithZipSupport.php

```diff
@@ -0,0 +1,111 @@
+use demosplan\DemosPlanCoreBundle\Logic\Statement\StatementCopier;
+use demosplan\DemosPlanCoreBundle\Logic\Statement\StatementService;
+use demosplan\DemosPlanCoreBundle\Logic\StatementAttachmentService;
+use demosplan\DemosPlanCoreBundle\Logic\User\CurrentUserService;
+use demosplan\DemosPlanCoreBundle\Logic\User\OrgaService;
+use demosplan\DemosPlanCoreBundle\Logic\ZipImportService;
+use Doctrine\ORM\EntityManagerInterface;
+use InvalidArgumentException;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webmozart\Assert\Assert;
+
+class StatementSpreadsheetImporterWithZipSupport extends StatementSpreadsheetImporter
+{
+    private ?array $fileMap;
+    public function __construct(
+        CurrentProcedureService $currentProcedureService,
+        CurrentUserService $currentUser,
+        ElementsService $elementsService,
+        OrgaService $orgaService,
+        StatementCopier $statementCopier,
+        StatementService $statementService,
+        TranslatorInterface $translator,
+        ValidatorInterface $validator,
+        private readonly ZipImportService $zipImportService,
+        private readonly FileService $fileService,
+        private readonly NCNameGenerator $nameGenerator,
+        private readonly EntityManagerInterface $entityManager,
+        private readonly StatementAttachmentService $statementAttachmentService
+    ) {
+        parent::__construct($currentProcedureService, $currentUser, $elementsService, $orgaService, $statementCopier, $statementService, $translator, $validator, $this->entityManager);
+    }
+
+    private function getStatementFromRowBuilder(StatementFromRowBuilder $baseStatementBuilder): StatementFromRowBuilderWithZipSupport
+    {
+        return new StatementFromRowBuilderWithZipSupport(
+            $this->validator,
+            $this->getFileMap(),
+            $this->fileService,
+            $this->entityManager,
+            $baseStatementBuilder,
+            $this->statementAttachmentService
+        );
+    }
+
+    protected function getColumnMapping(): array
+    {
+        [$baseColumns, $builder] = parent::getColumnMapping();
+        $builder = $this->getStatementFromRowBuilder($builder);
+        // add new columns
+        $baseColumns['Referenzen auf Anhänge'] = [$builder, 'setFileReferences'];
+        $baseColumns['Referenzen auf original Anhang'] = [$builder, 'setOriginalFileReferences'];
+
+        return [$baseColumns, $builder];
+    }
+
+    /**
+     * @throws UserNotFoundException
+     * @throws CopyException
+     * @throws UnexpectedWorksheetNameException
+     * @throws DemosException
+     * @throws StatementElementNotFoundException
+     * @throws MissingPostParameterException
+     * @throws InvalidDataException
+     */
+    public function process(SplFileInfo $zipFileInfo): void
```

Comment:
Variable name differs from parent

Reply:
Yes, the SpreadsheetZipImporter gets called first expecting a SplFileInfo of a zip file.
Then after extracting the files it calls its parent with the extracted SplFileInfo of the xlsx.
There the xlsx file gets processed as before - except the used builder being different.

Reply:
I was just referring to the fact that the name if the variable differs from the parent. Usually this should not be the case to avoid confusion. But no blocker here